### PR TITLE
pdctl: verify pd address

### PR DESCRIPTION
**Pull request by nolouch**
close https://github.com/pingcap/pd/issues/507
PTAL @siddontang @overvenus @huachaohuang

### CODE REVIEW

File: pdctl/command/global.go [hunk 1]
```diff
@@ -75,6 +84,25 @@ func printResponseError(r *http.Response) {
...
+	}
+	if u.Scheme == "" {
+		u.Scheme = "http"
+	}
+	addr := u.String()
+	reps, err := http.Get(fmt.Sprintf("%s/%s", addr, pingPrifix))
+	if err != nil {
+		return err
+	}
+	if reps.StatusCode != http.StatusOK {
```
Comment (siddontang) [inline pdctl/command/global.go]: must read  all response body and close

File: pdctl/ctl.go [hunk 2]
```diff
@@ -48,7 +48,10 @@ func Start(args []string) (string, error) {
 	rootCmd.SetArgs(args)
 	rootCmd.SilenceErrors = true
 	rootCmd.ParseFlags(args)
-	command.InitPDClient(rootCmd)
+	err := command.InitPDClient(rootCmd)
+	if err != nil {
+		return err.Error(), err
```
Comment (siddontang) [inline pdctl/ctl.go]: why return `err.Error()` here?

File: cmd/pd-ctl/main.go [hunk 3]
```diff
@@ -87,7 +87,9 @@ func loop() {
 			}
 			continue
 		}
-
+		if line == "exit" {
```
Comment (siddontang) [inline cmd/pd-ctl/main.go]: split it to another PR

File: pdctl/ctl.go (hunk 2 above)
Comment (nolouch) [inline pdctl/ctl.go]: this string will print.

File: cmd/pd-ctl/main.go (hunk 3 above)
Comment (nolouch) [inline cmd/pd-ctl/main.go]: if `InitPDClient` meet error, it will return error directly without execute the command. so I move `exit` to here.

File: pdctl/ctl.go (hunk 2 above)
Comment (siddontang) [inline pdctl/ctl.go]: why not check error and print the error directly?

File: pdctl/ctl.go [hunk 4]
```diff
@@ -44,14 +46,17 @@ func init() {
...
 // Start run Command
-func Start(args []string) (string, error) {
+func Start(args []string) {
 	rootCmd.SetArgs(args)
 	rootCmd.SilenceErrors = true
 	rootCmd.ParseFlags(args)
-	command.InitPDClient(rootCmd)
+	err := command.InitPDClient(rootCmd)
+	if err != nil {
+		fmt.Println(err)
```
Comment (siddontang) [inline pdctl/ctl.go]: should exit here directly?

**Comment by siddontang** [comment]
PTAL @disksing

File: pdctl/command/global.go [hunk 5]
```diff
@@ -111,6 +118,27 @@ func printResponseError(r *http.Response) {
...
+		u.Scheme = "http"
+	}
+	addr := u.String()
+	reps, err := http.Get(fmt.Sprintf("%s/%s", addr, pingPrifix))
+	if err != nil {
+		return err
+	}
+	defer reps.Body.Close()
+	if reps.StatusCode != http.StatusOK {
+		ioutil.ReadAll(reps.Body)
```
Comment (disksing) [inline pdctl/command/global.go]: You don't have to read it.

Comment (siddontang) [inline pdctl/command/global.go]: If not read, the connection FD will not close even you call body close later. I remember this is a pitfall before, but don't know go improves it or not.   

Btw, We should call `ioutil.ReadAll(reps.Body)`  before the status check.

**Comment by disksing** [comment]
LGTM
